File: Python/2-addTwoNumbers.py
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
    head = ListNode(l1.val + l2.val)
    cur = head
    while l1.next or l2.next:
        l1 = l1.next if l1.next else ListNode()
        l2 = l2.next if l2.next else ListNode()
        cur.next = ListNode(l1.val + l2.val + cur.val // 10)
        cur.val = cur.val % 10
        cur = cur.next

    if cur.val >= 10:
        cur.next = ListNode(cur.val // 10)
        cur.val = cur.val % 10

    return head

File: Campus/linkedList/2-Add Two Numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function (l1, l2) {
    let res = new ListNode(null);
    let nextRes = res;

    let carry = 0;
    let val = 0;
    while (l1 != null || l2 != null) {
        const x = (l1 != null) ? l1.val : 0;
        const y = (l2 != null) ? l2.val : 0;

        val = (x + y + carry) % 10;
        carry = Math.floor((x + y + carry) / 10);

        nextRes.next = new ListNode(val);
        nextRes = nextRes.nextß;

        if (l1 != null) l1 = l1.next;
        if (l2 != null) l2 = l2.next;
    }

    if (carry) nextRes.next = new ListNode(carry);

    return res.next;
};